packages/advanced-logic/src/extensions/payment-network/any-to-eth-proxy.ts

```diff
@@ -0,0 +1,119 @@
+import { ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import EthereumFeeProxyPaymentNetwork from './ethereum/fee-proxy-contract';
+import { supportedCurrencies } from './chainlink-supported-currencies';
+
+const CURRENT_VERSION = '0.1.0';
+// Default network if the storage data does not give any
+const DEFAULT_NETWORK = 'mainnet';
+
+export default class AnyToEthProxyPaymentNetwork extends EthereumFeeProxyPaymentNetwork {
+  public constructor(
+    extensionId: ExtensionTypes.ID = ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ETH_PROXY,
+    currentVersion: string = CURRENT_VERSION,
+  ) {
+    super(extensionId, currentVersion, Object.keys(supportedCurrencies));
+  }
+
+  /**
+   * Creates the extensionsData to create the extension ETH fee proxy contract payment detection
+   *
+   * @param creationParameters extensions parameters to create
+   *
+   * @returns IExtensionCreationAction the extensionsData to be stored in the request
+   */
+  public createCreationAction(
+    creationParameters: ExtensionTypes.PnAnyToEth.ICreationParameters,
+  ): ExtensionTypes.IAction {
+    const network = creationParameters.network;
+    if (!network) {
+      throw Error('network is required');
```

Comment:
Why don't we force this in the type instead ?

Reply:
It's a pain with the inheritance. It's the same kind of issue that need a proper schema validation as @benjlevesque suggests.